curation-migration-datasync/database/dataService.ts

```diff
@@ -130,6 +132,81 @@ export class DataService {
     });
   }
 
+  /**
+   * updates the curated_feed_prospects table and curated_feed_queued_items
+   * fields, if they are set in the eventBody.
+   * won't update the fields if they are set to null in the eventBody.
+   * @param eventBody event body
+   * @param curatedRecId curatedRecId corresponding to the approvedItem's externalId
+   * @param domainId domain id from parser, optional field set only when
+   *        publisher is not null in the event body.
+   */
+  public async updateApprovedItem(
+    eventBody: ApprovedItemPayload,
+    curatedRecId: number,
+    domainId?: string
+  ): Promise<void> {
+    const item = await this.db(config.tables.curatedFeedItems)
+      .select()
+      .join(
+        config.tables.curatedFeedProspects,
+        'curated_feed_prospects.prospect_id',
+        'curated_feed_items.prospect_id'
+      )
+      .join(
+        config.tables.curatedFeedQueuedItems,
+        'curated_feed_queued_items.queued_id',
+        'curated_feed_items.queued_id'
+      )
+      .where('curated_rec_id', curatedRecId)
+      .first();
+
+    const topDomainId = domainId
+      ? await this.fetchTopDomain(eventBody.url, domainId)
+      : item['domain_id'];
+
+    const curator = eventBody.createdBy
```

Comment:
note: we consider curator as the one who created, so update only if createdBy is updated.

ignore if updatedby is set.

Reply:
note: `updatedBy` is not included in the event payload